### PR TITLE
Update environment.yml and setup.py

### PR DESCRIPTION
**Pull request by mdpunch**
Have to pin numpy to 1.22, as @maxnoe remarks "ctapipe 0.12 depends on astropy 4, which is incompatible with the most recent numpy release 1.23"

### CODE REVIEW

File: environment.yml [hunk 1]
```diff
@@ -4,6 +4,7 @@ channels:
 dependencies:
   - python=3.8 # nail the python version, so conda does not try upgrading / downgrading
   - conda-forge::ctapipe=0.12
+  - numpy=1.22.4 # higher versions don't work due to astropy4 dependency in ctapipe0.12
```
Comment (maxnoe) [inline environment.yml:7]: I'd suggest not pinning to `1.22.4` as that would prevent updating to a possible 1.22.5 bug-fix release

**Comment by maxnoe** [comment]
Please also add it in `setup.py` (this is the actually important requirement, the env is just "nice-to-have".

File: environment.yml (hunk 1 above)
Comment (mdpunch) [reply environment.yml]: Yes, thanks.  Trying to modify the commit now...

**Comment by mdpunch** [comment]
Question for @jlenain , should these be in the `setup.py` of `nectarchain` also?
For the moment, there are no pinned versions there.

**Comment by mdpunch** [comment]
i.e.,
```
        'astropy~=4.2',
        'ctapipe~=0.12',
        'numpy~=1.22.4',
```

**Comment by maxnoe** [comment]
`nectarchain` should depend at least on `ctapipe_io_nectarcam`, right?

**Comment by mdpunch** [comment]
> `nectarchain` should depend at least on `ctapipe_io_nectarcam`, right?

I would have thought, but it doesn't seem to!

The installation of `ctapipe_io_nectarcam`  is a separate `git clone` and `pip install`.

I guess there are reasons...

**Comment by maxnoe** [comment]
The reason will be that `ctapipe_io_nectarcam` is not released to pypi, which should be done.

**Comment by jlenain** [comment]
> Question for @jlenain , should these be in the `setup.py` of `nectarchain` also? For the moment, there are no pinned versions there.

Hi @mdpunch , yes, please, also add the `numpy` pinned version to `setup.py`. Thanks!

File: environment.yml [hunk 2]
```diff
@@ -4,6 +4,7 @@ channels:
 dependencies:
   - python=3.8 # nail the python version, so conda does not try upgrading / downgrading
   - conda-forge::ctapipe=0.12
+  - numpy~=1.22.4 # higher versions >=1.23 don't work due to astropy4 dependency in ctapipe0.12
```
Comment (jlenain) [inline environment.yml:7]: Hi @mdpunch ,

As noted already by @maxnoe , the syntax for the environment specifications for `conda` does not follow the same as `setup.py`.
Please change:
```  - numpy~=1.22.4```
with:
```  - numpy=1.22```

**Review by jlenain — CHANGES_REQUESTED** [review]

**Review by mdpunch — COMMENTED** [review]
Done

File: environment.yml (hunk 2 above)
Comment (mdpunch) [reply environment.yml]: Done (I think)

**Comment by jlenain** [comment]
> Done

Thanks a lot, @mdpunch , merging now.

**Pull request merged**